src/app.rs

```diff
@@ -100,17 +115,12 @@ pub struct Bound {
     pub proxy_addresses: proxy::Addresses,
 
     pub shutdown: signal::Shutdown,
-    tasks: Vec<JoinHandle<()>>,
     config: config::Config,
     drain_tx: drain::Signal,
 }
 
 impl Bound {
     pub async fn spawn(self) -> anyhow::Result<()> {
```

Comment:
Given we already spawned the tasks and now this just waits for shutdown maybe we need to call this `await_termination` or something

Reply:
Will think about it